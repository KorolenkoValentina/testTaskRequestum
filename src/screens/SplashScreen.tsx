import React from 'react';
import { TouchableOpacity} from 'react-native';
import styled from 'styled-components/native';

import  Logo from '../image/logo.svg'
import { ButtonProps, NavigationRoutes, SplashScreenProps } from '../components/interfaces';

const Container = styled.View`
  flex: 1;
  justify-content: space-between;
  align-items: center;
`;

const BgImage = styled.Image`
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
`;

const LogoContainer = styled.View`
  margin-top: 200px; 
`;
const ButtonContainer = styled.View`
  width: 80%;
  margin-bottom: 50px; 
`;

const Button = styled(({ pink, ...props }) => <TouchableOpacity {...props} />)<ButtonProps>`
  background-color: ${({ pink, theme }) => pink ? theme.colors.pink : theme.colors.dark};
  padding-vertical: 15px;
  border-radius: 5px;
  margin-bottom: 15px;
  shadow-color: ${({ theme, pink }) => (pink ? theme.colors.pink : theme.colors.dark)};
  shadow-offset: 0px 12px; 
  shadow-opacity: 0.58; 
  shadow-radius: 16px; 
  elevation: 5; /* Elevation для Android */;
`;


const ButtonText = styled.Text`
  color: ${({ theme }) => theme.colors.white};
  font-size: 12px; 
  text-align: center;
`;

const SplashScreen: React.FC<SplashScreenProps> = ({ navigation }) => {
  return (
    <Container>
      <BgImage source={require('../image/bg.png')} />
        
      <LogoContainer>
        <Logo testID='logo'/>
      </LogoContainer>
      <ButtonContainer>
        <Button pink testID='loginButton' onPress={() => navigation.navigate(NavigationRoutes.LOG_IN)}>
          <ButtonText>Login</ButtonText>
        </Button>
        <Button testID='registerButton' onPress={() => navigation.navigate(NavigationRoutes.SIGN_UP)}>
          <ButtonText>Register</ButtonText>
        </Button>
      </ButtonContainer>
    </Container>
  );
};

export default SplashScreen;
