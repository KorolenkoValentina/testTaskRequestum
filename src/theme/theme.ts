const theme = {
    colors: {
        white : '#FFFFFF',
        lightGray: '#818181',
        pink: '#EB0057',
        dark: '#262727',
    },
    
};
  
export default theme;