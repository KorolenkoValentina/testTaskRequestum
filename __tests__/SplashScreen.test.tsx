
import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components/native'; 
import  SplashScreen from '../src/screens/SplashScreen'
import theme from '../src/theme/theme';


describe('SplashScreen', () => {
  const mockNavigation = {
    navigate: jest.fn(),
  };
 
  test('renders correctly', () => {
    const tree = renderer.create(
      <ThemeProvider theme={theme}> 
        <SplashScreen navigation={mockNavigation} />
      </ThemeProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('navigates to Login screen when Login button is pressed', () => {
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <SplashScreen navigation={mockNavigation} />
      </ThemeProvider>
    );
    const instance = component.root;

    const loginButton = instance.findByProps({ testID: 'loginButton' });
    loginButton.props.onPress();

    expect(mockNavigation.navigate).toHaveBeenCalledWith('Log in');
  });
  test('navigates to Register screen when Register button is pressed', () => {
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <SplashScreen navigation={mockNavigation} />
      </ThemeProvider>
    );
    const instance = component.root;

    const registerButton = instance.findByProps({ testID: 'registerButton' });
    registerButton.props.onPress();

    expect(mockNavigation.navigate).toHaveBeenCalledWith('Sign Up');
  });
  test('renders Logo component', () => {
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <SplashScreen navigation={mockNavigation} />
      </ThemeProvider>
    );
    const instance = component.root;

    const logo = instance.findByProps({ testID: 'logo' });

    expect(logo).toBeTruthy();
  });
});