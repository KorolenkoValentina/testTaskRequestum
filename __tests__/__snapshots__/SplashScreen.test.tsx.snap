// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplashScreen renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "justifyContent": "space-between",
    }
  }
>
  <Image
    source={1}
    style={
      {
        "height": "100%",
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
  />
  <View
    style={
      {
        "marginTop": 200,
      }
    }
  >
    <SvgMock
      testID="logo"
    />
  </View>
  <View
    style={
      {
        "marginBottom": 50,
        "width": "80%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#EB0057",
          "borderBottomLeftRadius": 5,
          "borderBottomRightRadius": 5,
          "borderTopLeftRadius": 5,
          "borderTopRightRadius": 5,
          "elevation": 5,
          "marginBottom": 15,
          "opacity": 1,
          "paddingVertical": 15,
          "shadowColor": "#EB0057",
          "shadowOffset": {
            "height": 12,
            "width": 0,
          },
          "shadowOpacity": 0.58,
          "shadowRadius": 16,
        }
      }
      testID="loginButton"
    >
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontSize": 12,
            "textAlign": "center",
          }
        }
      >
        Login
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#262727",
          "borderBottomLeftRadius": 5,
          "borderBottomRightRadius": 5,
          "borderTopLeftRadius": 5,
          "borderTopRightRadius": 5,
          "elevation": 5,
          "marginBottom": 15,
          "opacity": 1,
          "paddingVertical": 15,
          "shadowColor": "#262727",
          "shadowOffset": {
            "height": 12,
            "width": 0,
          },
          "shadowOpacity": 0.58,
          "shadowRadius": 16,
        }
      }
      testID="registerButton"
    >
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontSize": 12,
            "textAlign": "center",
          }
        }
      >
        Register
      </Text>
    </View>
  </View>
</View>
`;
